process/transaction/shardProcess.go

```diff
@@ -678,6 +665,8 @@ func (txProc *txProcessor) processUserTx(
 			err.Error())
 	}
 
+	// TODO: check why executeFailedRelay just for these 2 cases? What happens if moveBalance fails in the senderShard
```

Comment:
these 2 cases are when execution fails in sender shard.
In destination shard not Payable SC is resolved in another way through smart contract processing in case of relayed TX. smartContractProcessor.ProcessSmartContractResult -> moveBalance case.

Reply:
ok